### PR TITLE
feat!: Adds `Newsletter` and its child components

### PR DESCRIPTION
**Pull request by lucasfp13**
## What's the purpose of this pull request?

This PR intends to add the `Newsletter` and its child components to `@faststore/ui`.

**BREAKING CHANGE**: We are removing the `__experimentalNewsletterAddendum` override since this PR adds the `NewsletterAddendum` component to `@faststore/ui`.

## How to test it?

- Run `yarn workspace site dev` on `faststore` root directory and check the `Newsletter` page or use the [Doc Preview Link](https://faststore-site-git-feat-fs-247-move-newsletter-23852e-faststore.vercel.app/components/organisms/newsletter).

### Starters Deploy Preview

https://github.com/vtex-sites/starter.store/pull/346

### CODE REVIEW

File: packages/core/src/components/sections/Newsletter/Newsletter.tsx [hunk 1]
```diff
@@ -61,13 +79,15 @@ const Newsletter = function Newsletter({
...
+  subscribeButtonLoadingLabel,
   card,
   toastSubscribe,
   toastSubscribeError,
+  colorVariant,
   ...otherProps
 }: NewsletterProps) {
   return (
-    <Section className={`${styles.section} section-newsletter`}>
+    <Section className={`${styles.section} section-newsletter layout__section`}>
```
Comment (icazevedo) [inline packages/core/src/components/sections/Newsletter/Newsletter.tsx:90]: Why did you add this class?

File: packages/components/src/organisms/Newsletter/NewsletterAddendum.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  function NewsletterAddendum(
+    { addendum, testId = 'fs-newsletter-addendum', ...otherProps },
+    ref
+  ) {
+    return (
+      <span
+        ref={ref}
+        data-testid={testId}
+        data-fs-newsletter-addendum
+        dangerouslySetInnerHTML={{ __html: addendum }}
```
Comment (icazevedo) [inline packages/components/src/organisms/Newsletter/NewsletterAddendum.tsx:26]: I know this reflects the default behavior from core, but it comes out a little weird to me that this component expects the HTML value by default.

That is true only for our use, and I don't see other people using HTML for this component inside custom sections. I think they'd put the value directly, and for that using `children` is a better API IMO.

With `...otherProps` `@faststore/core` would be able to set the HTML the way we do here.

I think in this specific case it's close to a nitpick but in general terms the question is: should we "leak" the pattern of usage from `@faststore/core` to `@faststore/ui`?

File: packages/core/src/components/sections/Newsletter/Newsletter.tsx (hunk 1 above)
Comment (lucasfp13) [reply packages/core/src/components/sections/Newsletter/Newsletter.tsx:90]: I just replicated this class [based on the changes](https://github.com/vtex/faststore/pull/1902/files#diff-763c1ca044e8603b0d663a7764b60255ac3cfaf2a87f841f17846e6aad6d5723) that @renatamottam had already applied before.

**Review by renatamottam — APPROVED** [review]
Happy to finally see this improvement moving forward 🥳 
Thanks for all the work, @lucasfp13! this looks nice :)

File: packages/core/src/components/sections/Newsletter/Newsletter.tsx (hunk 1 above)
Comment (renatamottam) [reply packages/core/src/components/sections/Newsletter/Newsletter.tsx:90]: All our page sections have this class. It adds our default vertical margins.

**Review by icazevedo — APPROVED** [review]

File: apps/site/pages/components/organisms/newsletter.mdx [hunk 3]
```diff
@@ -0,0 +1,385 @@
+---
+description: The Newsletter component is a section with at least one input text with an action which its value could be submitted..
```
Comment (hellofanny) [inline apps/site/pages/components/organisms/newsletter.mdx:2]:
```suggestion
description: The Newsletter component is a section with at least one input text with an action allowing users to submit their input.
```

File: apps/site/pages/components/organisms/newsletter.mdx [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+    NewsletterForm: <PropsSection propsList={newsletterFormProps} />,
+    NewsletterHeader: <PropsSection propsList={newsletterHeaderProps} />,
+  }[component]
+}
+
+<header>
+
+# Newsletter
+
+The Newsletter component is a section with at least one input text with an action which its value could be submitted.
```
Comment (hellofanny) [inline apps/site/pages/components/organisms/newsletter.mdx:75]:
```suggestion
The Newsletter component is a section with at least one input text with an action allowing users to submit their input.
```

File: apps/site/pages/components/organisms/newsletter.mdx [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+  NewsletterForm,
+  NewsletterHeader,
+} from '@faststore/ui'
+```
+
+Import Styles
+
+```tsx
+import '@faststore/ui/src/components/organisms/Newsletter/styles.scss'
+```
```
Comment (hellofanny) [inline apps/site/pages/components/organisms/newsletter.mdx:97]:
```suggestion
### Import Styles into your FastStore project

To apply the styles of this component in your FastStore project, import the following into your stylesheet:

```scss copy
@import '@faststore/ui/src/components/organisms/Newsletter/styles.scss';
```

Follow the instructions in the [Importing FastStore UI component styles](/docs/customizing/importing-styles-to-faststore-ui-components) tutorial.
```

Comment (hellofanny) [reply apps/site/pages/components/organisms/newsletter.mdx:97]: Updating to our new format. (The suggestion tool is not working as expected here), but here is the suggestion:

### Import Styles into your FastStore project

To apply the styles of this component in your FastStore project, import the following into your stylesheet:

```scss copy
@import '@faststore/ui/src/components/organisms/Newsletter/styles.scss';
```

Follow the instructions in the [Importing FastStore UI component styles](/docs/customizing/importing-styles-to-faststore-ui-components) tutorial.

File: apps/site/pages/components/organisms/newsletter.mdx [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+#### Card
+
+<Tabs items={['Example', 'Code']} defaultIndex="0">
+  <Tab>
+    <OverviewSection>
+      <NewsletterUsage />
+    </OverviewSection>
+    <OverviewSection>
+      <NewsletterUsage card />
+    </OverviewSection>
```
Comment (hellofanny) [inline apps/site/pages/components/organisms/newsletter.mdx:215]:
```suggestion
    <OverviewSection>
      <NewsletterUsage card/>
    </OverviewSection>
    <OverviewSection>
      <NewsletterUsage />
    </OverviewSection>
```

Comment (hellofanny) [reply apps/site/pages/components/organisms/newsletter.mdx:215]: The examples and code were misplaced.

File: apps/site/pages/components/organisms/newsletter.mdx [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+  />
+  <TokenRow
+    token="--fs-newsletter-accent-text-color"
+    value="var(--fs-newsletter-light-text-color)"
+  />
+</TokenTable>
+
+---
+
+## Customization
```
Comment (hellofanny) [inline apps/site/pages/components/organisms/newsletter.mdx:369]:
```suggestion
## Customization

For further customization, you can use the following data attributes:
```

File: packages/components/src/organisms/Newsletter/Newsletter.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import React, { forwardRef } from 'react'
+import type { HTMLAttributes, DetailedHTMLProps } from 'react'
+
+type ColorVariant = 'main' | 'light' | 'accent'
+
+export interface NewsletterProps
+  extends DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
+  /**
+   * Enables the card Variant
```
Comment (hellofanny) [inline packages/components/src/organisms/Newsletter/Newsletter.tsx:9]:
```suggestion
   * Enables the card Variant.
```

File: packages/core/src/components/sections/Newsletter/Newsletter.tsx [hunk 9]
```diff
@@ -42,14 +52,22 @@ export interface NewsletterProps
...
 
-  toastSubscribe: SubscribeMessage
+  toastSubscribe?: SubscribeMessage
 
-  toastSubscribeError: SubscribeMessage
+  toastSubscribeError?: SubscribeMessage
+  /**
+   * Specifies the component's color variant combination.
+   */
+  colorVariant?: UINewsletterProps['colorVariant']
```
Comment (hellofanny) [inline packages/core/src/components/sections/Newsletter/Newsletter.tsx:70]: Just to improve the readability, can we reorder so UINewsletterProps can be listed together? 

privacyPolicy, card, colorVariant, subscribeButtonLabel...

File: packages/ui/src/components/organisms/BannerText/styles.scss [hunk 10]
```diff
@@ -114,7 +113,7 @@
     }
   }
 
-  &[data-fs-banner-color-variant="accent"] {
+  &[data-fs-banner-text-color-variant="accent"] {
```
Comment (hellofanny) [inline packages/ui/src/components/organisms/BannerText/styles.scss:116]: 👍

**Review by hellofanny — APPROVED** [review]
Thanks for updating this component! ⭐ 
Well done @lucasfp13, pre approving! I've made a few suggestions related to its documentation, can u pls take a look? 💯

**Comment by hellofanny** [comment]
@lucasfp13 just one more thing, this component is missing in this [list](https://faststore-site-eposxx1np-faststore.vercel.app/components) , can you also add to it [here](https://github.com/vtex/faststore/blob/56162750c26220e7dd94b21b92a4d094be70112a/apps/site/pages/components/index.mdx#L993) in the index pls. :)
